AlembicImporter/Assets/API_UnitTests/AlembicAPITest.cs

```diff
@@ -0,0 +1,109 @@
+        void TestAndScrub () {
+            Assert.That (Application.dataPath + "/API_UnitTests/TestAPI.abc", Is.Not.Null);
+            File.Delete (Application.dataPath + "/API_UnitTests/TestAPI.abc");
+        }
+        // Records a frame then stops the recorder
+        IEnumerator Record (AlembicRecorder recorderTarget) {
+            var result = recorderTarget.BeginRecording ();
+            Assert.That (result, Is.True);
+            recorderTarget.ProcessRecording ();
+            yield return null;
+            recorderTarget.EndRecording ();
+        }
+        // Test Capture Components
+        [UnityTest]
+        public IEnumerator PublicCaptureComponents () {
+            SceneLoader ("TestAPI");
+            using (AlembicRecorder recorder = new AlembicRecorder ()) {
+                var conf = SetUpExport (recorder);
+                recorder.settings.AssumeNonSkinnedMeshesAreConstant = false;
+                recorder.settings.CaptureMeshRenderer = true;
+                recorder.settings.CaptureCamera = true;
+                recorder.settings.MeshSubmeshes = true;
+                Record (recorder);
+            }
+            TestAndScrub ();
+            yield return null;
+        }
+        // Test AeConfig Settings
+        [UnityTest]
+        public IEnumerator PublicAeConfig () {
+            SceneLoader ("TestAPI");
+            using (AlembicRecorder recorder = new AlembicRecorder ()) {
+                var conf = SetUpExport (recorder);
+
+                conf.FrameRate = 24;
+                conf.ScaleFactor = 1;
+                conf.SwapHandedness = true;
+                Record (recorder);
+            }
+            TestAndScrub ();
+            yield return null;
+        }
+        // Test Archive types
+        [UnityTest]
+        public IEnumerator PublicAeArchiveType () {
+            SceneLoader ("TestAPI");
+            using (AlembicRecorder recorder = new AlembicRecorder ()) {
+                var conf = SetUpExport (recorder);
+                conf.ArchiveType = UnityEngine.Formats.Alembic.Sdk.aeArchiveType.Ogawa;
+                conf.ArchiveType = UnityEngine.Formats.Alembic.Sdk.aeArchiveType.HDF5;
+                Record (recorder);
+            }
+            TestAndScrub ();
+            yield return null;
+        }
+        // Test AeConfig set and load
+        [UnityTest]
+        public IEnumerator PublicAeConfigSaveLoad () {
+            SceneLoader ("TestAPI");
+            using (AlembicRecorder recorder = new AlembicRecorder ()) {
+                var conf = SetUpExport (recorder);
+                recorder.settings.Conf = conf;
+                Record (recorder);
+            }
+            TestAndScrub ();
+            yield return null;
+        }
+        // Test Dispose current recording
+        [UnityTest]
+        public IEnumerator PublicDispose () {
+            SceneLoader ("TestAPI");
+            using (AlembicRecorder recorder = new AlembicRecorder ()) {
```

Comment:
this will automatically call dispose at the end of the curly brace. Let's try to simplify this test a bit:

```
AlembicRecorder recorder;
using(recorder = new AlembicRecorder()){}
// try doing recorder.settings or something else, assert it gives an exception

recorder = new AlembicRecorder();
recorder.Dispose();
// try doing recorder.settings or something else, assert it gives an exception
```

We don't need to do the full export for this one.